go/logic/migrator.go

```diff
@@ -396,6 +396,10 @@ func (this *Migrator) handleCutOverResult(cutOverError error) (err error) {
 	// Only on error:
 
 	if this.migrationContext.TestOnReplica {
+		if err := this.hooksExecutor.onTestReplicaComplete(); err != nil {
```

Comment:
Whoops missed this one. Is this the correct spot? It greatly depends on what you want to achieve! 
The name `onTestReplicaComplete` is certainly not fitting to this place; the migration is not complete.

In fact, when the test-on-replica is complete, `gh-ost` just bails out without restarting the replication. So I do believe the normal "migration complete" hook should just work well for you. We use it that way, and just pass completely different hooks to the test-on-replica invocation.